### PR TITLE
Add bmrks extension

### PR DESCRIPTION
**Pull request by raunofreiberg**
## Description

This is a companion bookmarking utility for [(Basic) Bookmarks](https://bmrks.com). There is currently only one command:

### Bookmark

Captures the current URL, title, and favicon from the browser currently in focus and active, and saves to a selected remote group on [bmrks.com](https://bmrks.com).

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by peduarte** [comment]
We already talked about this on Slack, but let's try to make the extension more aligned/consistent with web app! 🚀

**Comment by peduarte** [comment]
@raunofreiberg feedack in slack

File: extensions/bmrks/src/create-bookmark.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      }
+    >
+      <Form.Dropdown id="groupId" title="Group" onChange={setActiveGroupId}>
+        {groups.map((group) => (
+          <Form.Dropdown.Item key={group.id} value={group.id} title={group.name} />
+        ))}
+      </Form.Dropdown>
+      <Form.Separator />
+      <Form.TextField id="value" autoFocus title="Link, color, or text" value={value} onChange={setValue} />
+      {valueIsUrl && <Form.TextField id="title" title="Link title" value={titleValue || ""} onChange={setTitleValue} />}
```
Comment (pernielsentikaer) [inline extensions/bmrks/src/create-bookmark.tsx:137]:
```suggestion
      {valueIsUrl && <Form.TextField id="title" title="Link Title" value={titleValue || ""} onChange={setTitleValue} />}
```

File: extensions/bmrks/src/create-bookmark.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        </ActionPanel>
+      }
+    >
+      <Form.Dropdown id="groupId" title="Group" onChange={setActiveGroupId}>
+        {groups.map((group) => (
+          <Form.Dropdown.Item key={group.id} value={group.id} title={group.name} />
+        ))}
+      </Form.Dropdown>
+      <Form.Separator />
+      <Form.TextField id="value" autoFocus title="Link, color, or text" value={value} onChange={setValue} />
```
Comment (pernielsentikaer) [inline extensions/bmrks/src/create-bookmark.tsx:136]:
```suggestion
      <Form.TextField id="value" autoFocus title="Input" value={value} placeholder="Link, color, or text" onChange={setValue} />
```
Could maybe be something else for title, not sure I like that one I made 😂

File: extensions/bmrks/src/search-bookmarks.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  const markdown =
+    error === "Invalid login credentials" ? error + ". Please open the preferences and try again." : error;
+
+  if (error) {
+    return (
+      <Detail
+        markdown={markdown}
+        actions={
+          <ActionPanel>
+            <Action title="Open Extension Preferences" onAction={openExtensionPreferences} />
```
Comment (pernielsentikaer) [inline extensions/bmrks/src/search-bookmarks.tsx:81]:
```suggestion
            <Action title="Open Extension Preferences" icon={Icon.Gear} onAction={openExtensionPreferences} />
```

File: extensions/bmrks/src/search-bookmarks.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    >
+      {bookmarks.length > 0 &&
+        bookmarks.map((bookmark) => {
+          const hostname = bookmark.type === "link" ? getHostname(bookmark.url || "") : "";
+          let icon;
+          if (bookmark.favicon_url) {
+            icon = bookmark.favicon_url;
+          }
+          if (bookmark.type === "color") {
+            icon = { source: Icon.CircleFilled, tintColor: bookmark.title };
```
Comment (pernielsentikaer) [inline extensions/bmrks/src/search-bookmarks.tsx:39]: Maybe add `adjustContrast: false` so it doesn't change the color

File: extensions/bmrks/src/create-bookmark.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      <Detail
+        markdown={markdown}
+        actions={
+          <ActionPanel>
+            <Action title="Open Extension Preferences" onAction={openExtensionPreferences} />
+          </ActionPanel>
+        }
+      />
+    );
+  }
```
Comment (pernielsentikaer) [inline extensions/bmrks/src/create-bookmark.tsx:159]: Maybe make a component since it's just multiple places, also add Icon.Gear 🙂

**Review by pernielsentikaer — COMMENTED** [review]
Just checked this, and it looks awesome. I have some initial feedback 🙂

File: extensions/bmrks/src/search-bookmarks.tsx (hunk 4 above)
Comment (peduarte) [reply extensions/bmrks/src/search-bookmarks.tsx:39]: Good shout 👌

File: extensions/bmrks/src/create-bookmark.tsx (hunk 2 above)
Comment (peduarte) [reply extensions/bmrks/src/create-bookmark.tsx:136]: Let me try it locally

**Pull request merged**